google/resource_dataflow_job.go

```diff
@@ -135,7 +138,7 @@ func resourceDataflowJob() *schema.Resource {
 				Type:             schema.TypeString,
 				Optional:         true,
 				ForceNew:         true,
-				DiffSuppressFunc: compareSelfLinkOrResourceName,
+				DiffSuppressFunc: resourceDataflowJobSubnetworkDiffSuppress,
```

Comment:
if you use the below change, you can just use [`compareResourceNames`](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/utils/self_link_helpers.go#L15). Honestly, I'd also just switch to using it anyways - the only difference is that it won't check the region for drift, but region is really determined by the dataflow job's region)

Reply:
I'm not too sure how I feel about this. It's true that the `subnetwork` must be in the same region as the zone that the job is running in ([source](https://cloud.google.com/dataflow/docs/guides/specifying-networks#subnetwork_parameter)), but wouldn't using `compareResourceNames()` mean that changes to the `subnetwork` field that don't involve name changes are effectively ignored? That is, would this be misleading to the user? (e.g. they changed the region of the `subnetwork`, but Terraform effectively ignores it and assumes the same subnetwork as before).

Reply:
@jcanseco I wanted to avoid extra diff suppress definitions if we can. 
It doesn't hurt to do the full comparison, so I'm ok with keeping this, but as other comment mentions, the custom regexp doesn't handle every way Terraform (or at least the Google Provider) allows most users to specify resources URLs.

My reasoning in this case is that dataflow job region should drive the subnetwork value, rather than trying to catch whether the subnet and job region supplied by the Terraform user are different. This drift would only happen if the API job region changed (because there's no way to change the subnet region only on the API) and the user updated their Job region in state/config to the correct value but not the subnetwork. Otherwise, we'd detect a drift in Job region and this drift wouldn't matter.